src/Sylius/Bundle/ApiBundle/Validator/Constraints/ChosenPaymentRequestActionEligibilityValidator.php

```diff
@@ -40,7 +42,20 @@ public function validate(mixed $value, Constraint $constraint): void
         /** @var ChosenPaymentRequestActionEligibility $constraint */
         Assert::isInstanceOf($constraint, ChosenPaymentRequestActionEligibility::class);
 
-        $gatewayFactoryCommandProvider = $this->getCommandProvider($value);
+        if (null === $value->action) {
+            return;
+        }
```

Comment:
A bit unrelated, but what do you think about introducing a dedicated QueryContextBuilder that'd take care of resolving the default action of `AddPaymentRequest` when none has been passed through the API?
We could then remove resolving from the handler and have it validated earlier.
Similar could be done regarding the default payload.